fastats/core/ast_transforms/processor.py

```diff
@@ -22,55 +23,81 @@ def __init__(self, top_level_func, overrides, replaced, new_funcs=None):
             pprint(ast.dump(new_tree))
 
-        code_obj = self.recompile(new_tree, '<fastats>', 'exec')
+        code_obj = recompile(new_tree, '<fastats>', 'exec')
 
         self.top_level_func.__code__ = code_obj
-        return jit(self.top_level_func)
-
-    def recompile(self, source, filename, mode, flags=0, privateprefix=None):
-        """
-        This is based on an ActiveState recipe by Oren Tirosh:
-        http://code.activestate.com/recipes/578353-code-to-source-and-back/
-
-        Recompiles output back to a code object.
-        Source may also be preparsed AST.
-        """
-        a = source
-        node = a.body[0]
-
-        c0 = compile(a, filename, mode, flags, True)
-
-        # This code object defines the function. Find the function's actual body code:
-        for c in c0.co_consts:
-            if not isinstance(c, CodeType):
-                continue
-            if c.co_name == node.name and c.co_firstlineno == node.lineno:
-                break
-        else:
-            raise TypeError('Function body code not found')
-
-        # Re-mangle private names:
-        if privateprefix is not None:
-            def fixnames(names):
-                isprivate = re.compile('^__.*(?<!__)$').match
-                return tuple(privateprefix + name if isprivate(name)
-                             else name for name in names)
-
-            c = CodeType(
-                c.co_argcount, c.co_nlocals, c.co_stacksize, c.co_flags,
-                c.co_code, c.co_consts, fixnames(c.co_names),
-                fixnames(c.co_varnames), c.co_filename, c.co_name,
-                c.co_firstlineno, c.co_lnotab, c.co_freevars, c.co_cellvars)
-        return c
+        return convert_to_jit(self.top_level_func)
+
+
+def recompile(source, filename, mode, flags=0):
+    """
+    This is based on an ActiveState recipe by Oren Tirosh:
+    http://code.activestate.com/recipes/578353-code-to-source-and-back/
+
+    Recompiles output back to a code object.
+    Source may also be preparsed AST.
+    """
+    node = source.body[0]
+
+    c0 = compile(source, filename, mode, flags, True)
+
+    # This code object defines the function. Find the function's actual body code:
+    for c in c0.co_consts:
+        if not isinstance(c, CodeType):
+            continue
+        if c.co_name == node.name and c.co_firstlineno == node.lineno:
+            break
+    else:
+        raise TypeError('Function body code not found')
+
+    return c
+
+
+def uncompile(c):
+    """ uncompile(codeobj) -> [source, filename, mode, flags, firstlineno, privateprefix] """
+    if c.co_name == '<lambda>':
+        raise TypeError('lambda functions not supported')
+    if c.co_filename == '<string>':
+        raise ValueError('code without source file not supported')
+
+    filename = inspect.getfile(c)
+    try:
+        lines, firstlineno = inspect.getsourcelines(c)
+    except IOError:
+        raise Exception('source code not available')
```

Comment:
`except IOError as err:` and `raise RuntimeError('source code not available') from err` might be more suitable

Reply:
hmm, i agree `raise ... from` is better here, but i don't like RuntimeError. Will have a think about this one.